src/runtime/policy/ingress-validation.ts

```diff
@@ -162,6 +162,40 @@ export class IngressValidation {
     return this.policies.some(policy => policy.targets.some(
       target => target.schemaName === type.getEntitySchema().name));
   }
+
+  // Get the max readable type for `typeName` according to the given `policies`.
+  static getMaxReadType(typeName: string, policies: Policy[]): Type|null {
+    const fields = {};
+    let type = null;
+    for (const policy of policies) {
+      for (const target of policy.targets) {
+        if (typeName === target.schemaName) {
+          type = target.type;
+          IngressValidation.mergeFields(fields, target.getRestrictedFields());
+        }
+      }
+    }
+    return type ? EntityType.make([typeName], fields, type.getEntitySchema()) : null;
+  }
+
+  private static mergeFields(fields: {}, newFields: {}) {
```

Comment:
all `fields` need to be converted to the FieldType class here and everywhere

Reply:
Confused about this comment. `fields` and `newFields` is a map from `string` to `FieldType` class.

Reply:
Maybe Maria is arguing for typing these variables as `Dictionary<FieldType>` and not `{}`?

Reply:
yes, what Piotr said, thanks.